source/slang/slang-ir-diff-call.cpp

```diff
@@ -0,0 +1,90 @@
+// slang-ir-diff-call.cpp
+#include "slang-ir-diff-call.h"
+
+#include "slang-ir.h"
+#include "slang-ir-insts.h"
+
+namespace Slang
+{
+
+struct DerivativeCallProcessContext
+{
+    // This type passes over the module and replaces
+    // derivative calls with the processed derivative 
+    // function.
+    //
+    IRModule*                       module;
+
+    bool processModule()
+    {
+        // Run through all the global-level instructions, 
+        // looking for callable blocks.
+        for (auto inst : module->getGlobalInsts())
```

Comment:
I recommend doing a general recursive walk instead of being specific onglobal functions. This will allow you to look into the bodies of generic functions as well. You can use the code in slang-ir-generics-lowering-context.h